### ISSUE TITLE
Selecting a large file hangs UI

### ISSUE THREAD

**Issue by hmmwhatsthisdo** [howeyc/crocgui#32]
Selecting a large file in the file-picker (when sending a file) causes the UI to hang for some period of time. This appears to be roughly linear to the size of the file - a 400MB file hung the UI for 3 or 4 seconds, an 8gb one has hung the UI for well over a minute.

I'm not sure if the app is reading files into memory at the time they're selected for some reason, but I would imagine this isn't desired behavior in favor of streaming files into memory as they're sent.

**Comment by howeyc** [comment]
Yeah, it's basically copying the files into the app directory so I can pass files to the croc go library. The croc libary expects files that it can find/stat/open, but android just gives me an opaque byte reading stream.

It is something that I know should be possible to fix, but I would need to change a lot to be able to pass "readers" to croc and I haven't got around to it yet.

**Comment by abakum** [comment]
IMHO I fix it https://github.com/abakum/crocgui/blob/a72514e56f3edd36e28d60904b92c80fffbe3881/crocgui.apk
Now you can select the next file while the previous one is being loaded into the cache. And you can also cancel the caching.
Files queued for sending or received but not yet saved are now preserved across application restarts.
